code/modules/day_night/day_night_controller.dm

```diff
@@ -0,0 +1,253 @@
+/**
+ * Day Night Presets are easy to use day/night cycles for certain maps, they are what is used to dictate the sky color.
+ *
+ * IMPORTANT: The lightzones in this MUST all add up to 23, you cannot miss an hour!
+ */
+/datum/day_night_controller
+	/// The maps that this preset should be loaded on
+	var/load_map_name
+	/// The lightzones that this preset should have
+	var/list/lightzones = list()
+	/// The lightzones that this preset has loaded
+	var/list/lightzone_cache = list()
+	/// A list of loaded areas, this is an assosicative list witht he area appearance attached.
+	var/list/area_cache = list()
+	/// A list of areas that are not affected directly, but still have turf adjacency
+	var/list/unaffected_area_cache = list()
+	/// The affected Z level - Ideally, we will eventually be using a more robust system for loaded planets and such.
+	var/affected_z_level
+	/// Our current luminosity
+	var/current_luminosity = FALSE
+	/// Lookup table for the colors for each hour, 24 hour format starting at 0.
+	var/list/color_lookup_table = list()
+	/// Lookup table for the alpha values for each hour, 24 hour format starting at 0.
+	var/list/alpha_lookup_table = list()
+	/// Our currently set light color
+	var/current_light_color
+	/// Our currently set light alpha
+	var/current_light_alpha
+
+
+/datum/day_night_controller/New(incoming_affected_z_level)
+	if(!incoming_affected_z_level)
+		return
+	affected_z_level = incoming_affected_z_level
+	get_affected_areas()
+	load_lightzones()
+	compile_transitions()
+
+/**
+ * Simple proc to get the areas that are affected by this controller
+ */
+/datum/day_night_controller/proc/get_affected_areas()
```

Comment:
name is a tad weird, get implies a return

Reply:
will rename, renamed